redwood-protocol-host/src/commonMain/kotlin/app/cash/redwood/protocol/host/ProtocolBridge.kt

```diff
@@ -57,22 +57,26 @@ public class ProtocolBridge<W : Any>(
   )
   private val changedWidgets = mutableSetOf<ChangeListener>()
 
+  private val removeNodeById: (Id) -> Unit = { nodes.remove(it) }
+  private val poolIfReusable: (ProtocolNode<W>) -> Unit = { it ->
+    if (it.reuse) {
+      pool.addLast(it)
+    }
+  }
```

Comment:
Is this JavaScript? What a strange way to declare a function

```suggestion
  private fun poolIfReusable(node: ProtocolNode<W>) {
    if (node.reuse) {
      pool.addLast(node)
    }
  }
```

Reply:
(I assume you’ve got a reason for this, but as a reader of this code I want an explanation comment here where the weirdness is)

Reply:
If you declare it as a function then an additional class needs to be generated to map it back to a lambda type, and that class needs instantiated for each remove change. The current setup is allocation-free.

Reply:
I rolled all this back and will address separately. There's also a problem still on the guest-side. The lambda type is `KFunction1` which still causes `Id` to box. We probably need a `fun interface IdVisitor { fun visitId(id: Id) }` to avoid that on both guest and hots. Future problem.